pkg/execute/fake_kc_guard.go

```diff
@@ -0,0 +1,71 @@
+package execute
+
+// FakeCommandGuard provides functionality to resolve correlations between kubectl verbs and resource types.
+// TODO: Currently, we use dumb implementation that handles only a happy path, just for test purposes.
+//
+//	This will be replaced by https://github.com/kubeshop/botkube/issues/786.
+type FakeCommandGuard struct{}
+
+// Resource holds additional details about the K8s resource type.
+type Resource struct {
+	Name                    string
+	Namespaced              bool
+	SlashSeparatedInCommand bool
+}
+
+// GetAllowedResourcesForVerb returns allowed resources types for a given verb.
+func (f *FakeCommandGuard) GetAllowedResourcesForVerb(selectedVerb string, allConfiguredResources []string) ([]Resource, error) {
+	_, found := resourcelessVerbs[selectedVerb]
+	if found {
+		return nil, nil
+	}
+
+	// special case for 'logs'
+	if selectedVerb == "logs" {
+		return []Resource{
+			staticResourceMapping["deployments"],
+			staticResourceMapping["pods"],
+		}, nil
+	}
+
+	var out []Resource
+	for _, name := range allConfiguredResources {
+		out = append(out, staticResourceMapping[name])
+	}
+	return out, nil
+}
+
+// GetResourceDetails returns resource details.
+func (f *FakeCommandGuard) GetResourceDetails(verb, resourceType string) Resource {
```

Comment:
I see one problem here, but please ignore if @pkosiec will handle that with non-fake mapping.
- In `staticResourceMapping`, there are missing resources like `pv`, `pvc`, etc.
- Let say that, I override helm values yaml for `commands.resources` and add `pvc`
- Then I will get a nil pointer on the places where this function is used like [here](https://github.com/mszostok/botkube/blob/330c1073619368d3d44985abfe0756d39b32af54/pkg/execute/kubectl_survey.go#L212)

Reply:
yes, this is a dummy implementation as stated in the comment on top. I did it only to showcase how dropdowns are added/deleted based on selected options. I didn't want to spend too much time on it, so as a result it has such problems, this service is even not unit-tested. I expect that it will be replaced by Paweł's implementation at the beginning of the next week.